Crash on MacOS Sequioa 15.0 beta (24A5320a) from 13-08-2024

### Description

Since today, after installing the latest Sequioa beta, colima crashes on start. Previous beta worked fine.

```
$ colima start
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x50 pc=0x102fd4d28]

goroutine 1 [running]:
github.com/abiosoft/colima/cmd.prepareConfig(0x10329ee20)
	github.com/abiosoft/colima/cmd/start.go:455 +0xa88
github.com/abiosoft/colima/cmd.init.func18(0x10329ee20, {0x10331bb80?, 0x0?, 0x102fdc1c4?})
	github.com/abiosoft/colima/cmd/start.go:96 +0x70
github.com/spf13/cobra.(*Command).execute(0x10329ee20, {0x10331bb80, 0x0, 0x0})
	github.com/spf13/cobra@v1.8.1/command.go:970 +0x7e8
github.com/spf13/cobra.(*Command).ExecuteC(0x1032a0240)
	github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/abiosoft/colima/cmd/root.Execute()
	github.com/abiosoft/colima/cmd/root/root.go:68 +0x24
main.main()
	github.com/abiosoft/colima/cmd/colima/main.go:12 +0x1c
```

I was using vz, but reset config by using 'colima delete' and 'colima start'. The colima delete did work, but start still crashes.

Config:
```
cpu: 2
disk: 100
memory: 2
arch: aarch64
runtime: docker
hostname: colima
kubernetes:
  enabled: false
  version: v1.30.0+k3s1
  k3sArgs:
    - --disable=traefik
autoActivate: true
network:
  address: false
  dns: []
  dnsHosts: {}
forwardAgent: false
docker: {}
vmType: qemu
rosetta: false
mountType: sshfs
mountInotify: true
cpuType: ""
provision: []
sshConfig: true
sshPort: 0
mounts: []
env: {}
```

After deleting the colima.yaml configuration file, I tried colima start again:

```
$ colima start
INFO[0000] starting colima
INFO[0000] runtime: docker
INFO[0001] creating and starting ...                     context=vm
INFO[0001] downloading disk image ...                    context=vm
> QEMU binary "/opt/homebrew/bin/qemu-system-aarch64" seems properly signed with the "com.apple.security.hypervisor" entitlement
> "Attempting to download the image" arch=aarch64 digest= location=/Users/xxx/Library/Caches/colima/caches/fc0bfb9256e1f61294f2c41fc95ff4feb6c0daafccfb3b435e2d972e39bc8432.raw
> Downloaded the image from "/Users/xxx/Library/Caches/colima/caches/fc0bfb9256e1f61294f2c41fc95ff4feb6c0daafccfb3b435e2d972e39bc8432.raw"
> [hostagent] hostagent socket created at /Users/xxx/.colima/_lima/colima/ha.sock
> [hostagent] Using system firmware ("/opt/homebrew/share/qemu/edk2-aarch64-code.fd")
> exiting, status={Running:false Degraded:false Exiting:true Errors:[] SSHLocalPort:0} (hint: see "/Users/xxx/.colima/_lima/colima/ha.stderr.log")
FATA[0021] error starting vm: error at 'creating and starting': exit status 1

$ colima start
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x50 pc=0x1028c0d28]

goroutine 1 [running]:
github.com/abiosoft/colima/cmd.prepareConfig(0x102b8ae20)
	github.com/abiosoft/colima/cmd/start.go:455 +0xa88
github.com/abiosoft/colima/cmd.init.func18(0x102b8ae20, {0x102c07b80?, 0x0?, 0x1028c81c4?})
	github.com/abiosoft/colima/cmd/start.go:96 +0x70
github.com/spf13/cobra.(*Command).execute(0x102b8ae20, {0x102c07b80, 0x0, 0x0})
	github.com/spf13/cobra@v1.8.1/command.go:970 +0x7e8
github.com/spf13/cobra.(*Command).ExecuteC(0x102b8c240)
	github.com/spf13/cobra@v1.8.1/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/abiosoft/colima/cmd/root.Execute()
	github.com/abiosoft/colima/cmd/root/root.go:68 +0x24
main.main()
	github.com/abiosoft/colima/cmd/colima/main.go:12 +0x1c
```

It seems to crash on the line with "MacOS13OrNewer" in start.go (line 455):
```
	if util.MacOS() {
		if !cmd.Flag("network-address").Changed {
			startCmdArgs.Network.Address = current.Network.Address
		}
		if util.MacOS13OrNewer() {    // <---- here
			if !cmd.Flag("vm-type").Changed {
				startCmdArgs.VMType = current.VMType
			}
		}
		if util.MacOS13OrNewerOnArm() {
			if !cmd.Flag("vz-rosetta").Changed {
				startCmdArgs.VZRosetta = current.VZRosetta
			}
		}
		if util.MacOS15OrNewer() {
			if !cmd.Flag("nested-virtualization").Changed {
				startCmdArgs.NestedVirtualization = current.NestedVirtualization
			}
		}
	}
```

### Version

```
$ uname -a
Darwin MacBook-Pro-xxx.local 24.0.0 Darwin Kernel Version 24.0.0: Wed Aug  7 03:09:20 PDT 2024; root:xnu-11215.1.9~22/RELEASE_ARM64_T6000 arm64
$ colima version && limactl --version && qemu-img --version
colima version 0.7.1
git commit: ce7b155dbeb7843ac0d9966a6ad19dc8e3b56bb8
limactl version 0.22.0
qemu-img version 9.0.2
```

### Operating System

- [ ] macOS Intel <= 13 (Ventura)
- [ ] macOS Intel >= 14 (Sonoma)
- [ ] Apple Silicon <= 13 (Ventura)
- [X] Apple Silicon >= 14 (Sonoma)
- [ ] Linux

### Output of `colima status`

```
$ colima status
FATA[0000] colima is not running
```

### Reproduction Steps

1. Update to latest beta of Sequioa
2. colima start
3. crash


### Expected behaviour

_No response_

### Additional context

_No response_

--- a/cmd/start.go
+++ b/cmd/start.go
@@ -171,7 +171,7 @@ func init() {
 		}
 
 		// nested virtualization
-		if util.M3() && util.MacOS15OrNewer() {
+		if util.MacOSNestedVirtualizationSupported() {
 			startCmd.Flags().BoolVarP(&startCmdArgs.NestedVirtualization, "nested-virtualization", "z", false, "enable nested virtualization")
 			startCmd.Flag("nested-virtualization").Hidden = true
 		}
@@ -457,7 +457,7 @@ func prepareConfig(cmd *cobra.Command) {
 				startCmdArgs.VZRosetta = current.VZRosetta
 			}
 		}
-		if util.MacOS15OrNewer() {
+		if util.MacOSNestedVirtualizationSupported() {
 			if !cmd.Flag("nested-virtualization").Changed {
 				startCmdArgs.NestedVirtualization = current.NestedVirtualization
 			}

--- a/environment/vm/lima/yaml.go
+++ b/environment/vm/lima/yaml.go
@@ -44,7 +44,7 @@ func newConf(ctx context.Context, conf config.Config) (l limaconfig.Config, err 
 			}
 		}
 
-		if util.MacOS15OrNewer() {
+		if util.MacOSNestedVirtualizationSupported() {
 			l.NestedVirtualization = conf.NestedVirtualization
 		}
 	}

--- a/util/macos.go
+++ b/util/macos.go
@@ -29,6 +29,11 @@ func MacOS13OrNewer() bool { return minMacOSVersion("13.0.0") }
 // MacOS15OrNewer returns if the current OS is macOS 15 or newer.
 func MacOS15OrNewer() bool { return minMacOSVersion("15.0.0") }
 
+// MacOSNestedVirtualizationSupported returns if the current device supports nested virtualization.
+func MacOSNestedVirtualizationSupported() bool {
+	return M3() && MacOS15OrNewer()
+}
+
 func minMacOSVersion(version string) bool {
 	if !MacOS() {
 		return false

